File: Json/src/ec/edu/espe/json/model/ProductModel.java
package ec.edu.espe.json.model;

import java.util.ArrayList;

/**
 * Arelys
 */
public class ProductModel {

    private ArrayList<Product> products;
    private int autoId;

    public ProductModel() {
        products = new ArrayList<>();
        autoId = 1;
    }

    public ArrayList<Product> getAll() {
        return products;
    }

    public void addProduct(String name, float price) {
        Product p = new Product(autoId, name, price);
        products.add(p);
        autoId++;
    }

    public Product findById(int id) {
        for (Product p : products) {
            if (p.getId() == id) return p;
        }
        return null;
    }

    public boolean delete(int id) {
        Product p = findById(id);
        if (p != null) {
            products.remove(p);
            return true;
        }
        return false;
    }
}

File: EjemplosJson/CRUDPythonProducts/Product_model.py
class Product:
    def __init__(self, id=None, name=None, price=0, quantity=0):
        if id is not None and id != "":
            self.id = int(id)
        else:
            self.id = 0

        self.name = name
        self.price = float(price)
        self.quantity = int(quantity)
